aea/configurations/base.py

```diff
@@ -955,10 +952,6 @@ class SkillConfig(ComponentConfiguration):
     FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset(
         ["behaviours", "handlers", "models", "is_abstract", "build_directory"]
     )
-    FIELDS_WITH_NESTED_FIELDS: FrozenSet[str] = frozenset(
-        ["behaviours", "handlers", "models"]
-    )
-    NESTED_FIELDS_ALLOWED_TO_UPDATE: FrozenSet[str] = frozenset(["args"])
```

Comment:
The reason we had this was that within the below, only fields nested inside `args` are allowed to be updated:

```
models:
  default_dialogues:
    args: {}
    class_name: DefaultDialogues
  fipa_dialogues:
    args: {}
    class_name: FipaDialogues
  ledger_api_dialogues:
    args: {}
    class_name: LedgerApiDialogues
  oef_search_dialogues:
    args: {}
    class_name: OefSearchDialogues
  signing_dialogues:
    args: {}
    class_name: SigningDialogues
  strategy:
    args:
      is_ledger_tx: true
      location:
        latitude: 51.5194
        longitude: 0.127
      max_negotiations: 1
      max_quantity: 100
      max_tx_fee: 1
      max_unit_price: 20
      min_quantity: 1
      search_query:
        constraint_type: ==
        search_key: seller_service
        search_value: generic_service
      search_radius: 5.0
      service_id: generic_service
      stop_searching_on_result: true
    class_name: GenericStrategy
```

Reply:
We only have 2 parameters for models/behaviours/handlers, `args` and `class_name`. From which the we only want the `args` to be overridable but this should not be the reason for introducing a class variable in the base class, we should be able to filter out this one parameter in the `get_overridables` method rather than introducing a variable in the base class which is not even useful any other classes than `SkillConfig`.